gcsstore.go

```diff
@@ -420,6 +420,7 @@ func (o *gcsFSObject) Close() error {
 	if !o.opened {
 		return nil
 	}
+	defer os.Remove(o.cachepath)
```

Comment:
If o.cachedcopy.Sync() or o.cached.Close() fail, don't we still want to clean up the cachepath file, and actually cleanup everything else?

Or do you think, in those cases we'd want to keep the cached file around longer?  

Seems like if Close() is closed we should cleanup everything regardless of where the error happened, so maybe:
```go
func (o *gcsFSObject) Close() error {
	if !o.opened {
		return nil
	}
	defer func() {
		os.Remove(o.cachepath)
		o.cachedcopy = nil
		o.opened = false
	}()

	serr := o.cachedcopy.Sync()
	cerr := o.cachedcopy.Close()
	if serr != nil || cerr != nil {
		return fmt.Errorf("error on sync and closing localfile. %s sync=%v, err=%v", o.cachepath, serr, cerr)
	}

	if o.opened && !o.readonly {
		err := o.Sync()
		if err != nil {
			return err
		}
	}

	return nil
}
```